app/cart/controller/cartControllers.js

```diff
@@ -157,15 +155,13 @@ class CartController {
 
   userPreviousOrders = async (req, res) => {
     try {
-      const productsDetail = await Cart.find(
-        {
-          $match: { userId: req.params.id },
-        },
-        { products: 1, _id: 0 }
-      ).populate({ path: "products.productId" });
-      return res.status(200).json({
-        products: productsDetail,
-      });
+      const productsDetail = await Cart.find({
+        $match: { userId: req.params.id },
+      }).populate({ path: "products.productId" });
```

Comment:
ye line kis liye add ki hai

Reply:
"products" is a array, stored in the database containing "productID" ObjectID and "Quantity" Number. Many productID ko use kartay how within query he data populate karlia using .populate(). maray liye easy ho gaya front end pay data display karna.

Reply:
good